### ISSUE TITLE
role-name suppression not working

### ISSUE THREAD

**Issue by tumbl3w33d** [ansible/ansible-lint#2576]
##### Summary

There might be a connection with me using `meta-no-info` as well, i.e., not having `galaxy_info` in `meta/main.yml`, but I cannot suppress the `role-name` detection. The error is also shown without `meta-no-info`, but there would be a follow-up error once the suppression of `role-name` works, so I include the `meta-no-info` suppression here as well.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible-lint 6.8.0 using ansible 2.13.4
```
- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

`5.15.71-1-MANJARO #1 SMP PREEMPT Wed Sep 28 11:24:27 UTC 2022`

##### STEPS TO REPRODUCE

Create a `meta/main.yml` that only uses `dependencies`, no `galaxy_info`
``` yml
dependencies:
  - foo
```
Add suppression to `.ansible-lint`
```
skip_list:
  - role-name
  - meta-no-info
```

Run ansible lint.

##### Desired Behavior

No error/warning.

##### Actual Behavior

```paste below
ERROR    Computed fully qualified role name of playground does not follow current galaxy requirements.
Please edit meta/main.yml and assure we can correctly determine full role name:

galaxy_info:
role_name: my_name  # if absent directory name hosting role is used instead
namespace: my_galaxy_namespace  # if absent, author is used instead

Namespace: https://galaxy.ansible.com/docs/contributing/namespaces.html#galaxy-namespace-limitations
Role: https://galaxy.ansible.com/docs/contributing/creating_role.html#role-names

As an alternative, you can add 'role-name' to either skip_list or warn_list.

Computed fully qualified role name of playground does not follow current galaxy requirements.
Please edit meta/main.yml and assure we can correctly determine full role name:

galaxy_info:
role_name: my_name  # if absent directory name hosting role is used instead
namespace: my_galaxy_namespace  # if absent, author is used instead

Namespace: https://galaxy.ansible.com/docs/contributing/namespaces.html#galaxy-namespace-limitations
Role: https://galaxy.ansible.com/docs/contributing/creating_role.html#role-names

As an alternative, you can add 'role-name' to either skip_list or warn_list.
```

**Comment by ssbarnea** [comment]
I will look at the bug, maybe I missed something.

Still, you should know that github puts redirects every time you rename a repository, so everyone using the old repo name will still get the code. The redirects also work for git clones! We renamed some repositories multiple times and all the old names continued to work, i was really pleasantly surprised how well them implemented that.

**Comment by ssbarnea** [comment]
Both of these requirements are many years old and the fact that you can still use them with Ansible does not mean the linter would allow you to do so.

I will need to check with the rest of the team on this subject as we might decide to no longer support invalid namespaces and invalid role names. Our time is quite limited and we do want to improve the experience (or remove) for those that are following the current practices. Still, if you happen to make a pull-request to fix this that does to break something else, I will be glad to include it.

I will keep it open until we make a decision on it.

PS. Next week there is a talk at https://hackmd.io/@ansible-community/cs202210-planning that you might be interested about, look for "GalaxyNG Beta".

**Comment by tumbl3w33d** [comment]
I don't have any connection to galaxy and I know your view on it, because we already talked about that point here: https://github.com/ansible/schemas/issues/163

What I don't get is, why the linter should force this on me without any way to opt out although it seems to be a valid use case to just couple roles using `dependencies`.

**Comment by ssbarnea** [comment]
The main reason for this is related to how Ansible works internally and finds packages, regardless how you ship/install/deploy them, with or without a galaxy, rpms. etc.

**Comment by zeitounator** [comment]
@ssbarnea although I'm not really happy with your response, I understand it and it perfectly make sense. I'll disable ansible-lint for now and see if I have any time to dig into the code to propose an acceptable PR. But my overall impression is that my time would be better invested in transforming my project into a correctly named collection anyway...

Talking about this, my main concern is that the role has ~700k downloads and is still quite used. Moving along will break the actual uri people are used to (putting aside the download counter which will start over but that's mainly a problem for my ego :)). Is there any overall good practice on galaxy I can follow to make this process as smooth as possible for the users? Thanks in advance if you can point me to any good reference on the subject.

And since I don't have many occasions: thanks for your time and your excellent work, and also for the talk link. Cheers.

**Comment by ssbarnea** [comment]
Few years back, I made a chimera repository that was still a role and also a collection, mainly using symlinks. If you do not have modules or you only have standalone modules inside that might be possible.

Still, there is another thing to consider: maybe you should just create an "old" branch and ask anyone that wants to use your code to use switch. Switching from calling a role to calling a role form a collection is like two lines, one to call it with full name, and another one to add the collection to `requirements.yml`.

I would not put much value in download counters, you only need one or two people using the role in automation and you might believe your creation is extremely popular... among bots and scheduled tasks? github starts are bit better IMHO, but even this could tell the wrong story. I know some old projects with lots of stars, which are not well maintained and where much better alternative are available, even if they have like 10x less stars.

**Comment by bviktor** [comment]
I don't quite understand all the rambling about whether it makes sense to do things this way or that way. That's perfectly relevant. ansible-lint is a linter, not a coding coach, and I expect it to adhere to the configuration I feed it with.

What is certain is that:

- Namespaces make no sense in Galaxy roles, because it is inherited from your Galaxy (and thus your GitHub) username. You don't have to agree with this behavior, I don't agree with it either, but that won't make the issue go away, so forcing people to set up namespaces in the code only to be ignored by Galaxy will at best cause confusion.
- The one and only purpose of `skip_list` is to skip certain kinds of errors, and as of now, it does not fulfill that purpose, apparently it deliberately ignores it for random cherry picked directives. It makes zero sense, and this behavior will not make me specify made-up namespaces in my meta files, the only thing it might achieve is me dropping the use of ansible-lint altogether.
